platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/persistence/felix/FelixPersistenceStrategy.java

```diff
@@ -17,23 +17,94 @@
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.util.Dictionary;
+import java.util.Enumeration;
 
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
 import org.apache.felix.cm.file.ConfigurationHandler;
+import org.apache.felix.utils.properties.Properties;
 import org.codice.ddf.configuration.persistence.PersistenceStrategy;
+import org.codice.ddf.migration.ExportMigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * Class that persists configuration properties using the Felix' file format.
+ * Class that persists configuration properties using the Felix file formats (cfg and config).
  */
 public class FelixPersistenceStrategy implements PersistenceStrategy {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(FelixPersistenceStrategy.class);
+
+    private static final String FELIX_FILEINSTALL_FILENAME = "felix.fileinstall.filename";
+
+    private static final String CFG_EXTENSION = "cfg";
+
+    private static final String CONFIG_EXTENSION = "config";
+
     @Override
     public void write(OutputStream outputStream, Dictionary<String, Object> properties)
             throws IOException {
         notNull(outputStream, "OutputStream cannot be null");
         notNull(properties, "Properties cannot be null");
 
+        String fileName = getFileName(properties);
+        String fileExtension = FilenameUtils.getExtension(fileName);
+
+        if (isConfigFormat(fileExtension)) {
+            writeInConfigFormat(outputStream, properties);
+        } else if (isCfgFormat(fileExtension)) {
+            writeInCfgFormat(outputStream, properties);
+        } else if (defaultToConfigFormat(fileExtension)) {
```

Comment:
If we don't find a _felix.fileinstall.filename_ in a configuration in Config Admin, don't we want to default to a _.config_ file format?